src/main/java/com/englishtown/bitbucket/hook/MirrorRepositoryHook.java

```diff
@@ -1,262 +1,144 @@
 package com.englishtown.bitbucket.hook;
 
-import com.atlassian.bitbucket.hook.repository.AsyncPostReceiveRepositoryHook;
-import com.atlassian.bitbucket.hook.repository.RepositoryHookContext;
-import com.atlassian.bitbucket.i18n.I18nService;
-import com.atlassian.bitbucket.repository.RefChange;
+import com.atlassian.bitbucket.concurrent.BucketedExecutor;
+import com.atlassian.bitbucket.concurrent.BucketedExecutorSettings;
+import com.atlassian.bitbucket.concurrent.ConcurrencyPolicy;
+import com.atlassian.bitbucket.concurrent.ConcurrencyService;
+import com.atlassian.bitbucket.hook.repository.PostRepositoryHook;
+import com.atlassian.bitbucket.hook.repository.PostRepositoryHookContext;
+import com.atlassian.bitbucket.hook.repository.RepositoryPushHookRequest;
 import com.atlassian.bitbucket.repository.Repository;
-import com.atlassian.bitbucket.repository.RepositoryService;
-import com.atlassian.bitbucket.scm.CommandExitHandler;
-import com.atlassian.bitbucket.scm.DefaultCommandExitHandler;
-import com.atlassian.bitbucket.scm.ScmCommandBuilder;
-import com.atlassian.bitbucket.scm.ScmService;
-import com.atlassian.bitbucket.scm.git.command.GitScmCommandBuilder;
-import com.atlassian.bitbucket.setting.RepositorySettingsValidator;
+import com.atlassian.bitbucket.scm.git.GitScm;
+import com.atlassian.bitbucket.scope.RepositoryScope;
+import com.atlassian.bitbucket.scope.Scope;
+import com.atlassian.bitbucket.scope.ScopeVisitor;
+import com.atlassian.bitbucket.server.ApplicationPropertiesService;
 import com.atlassian.bitbucket.setting.Settings;
 import com.atlassian.bitbucket.setting.SettingsValidationErrors;
-import com.atlassian.sal.api.pluginsettings.PluginSettings;
-import com.atlassian.sal.api.pluginsettings.PluginSettingsFactory;
-import com.google.common.base.Strings;
+import com.atlassian.bitbucket.setting.SettingsValidator;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import javax.annotation.Nonnull;
-import java.lang.reflect.Method;
 import java.net.URI;
-import java.net.URISyntaxException;
-import java.util.*;
-import java.util.concurrent.ScheduledExecutorService;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
-public class MirrorRepositoryHook implements AsyncPostReceiveRepositoryHook, RepositorySettingsValidator {
-
-    protected static class MirrorSettings {
-        String mirrorRepoUrl;
-        String username;
-        String password;
-        String suffix;
-        String refspec;
-        boolean tags;
-        boolean notes;
-        boolean atomic;
-    }
+public class MirrorRepositoryHook implements PostRepositoryHook<RepositoryPushHookRequest>, SettingsValidator {
```

Comment:
I may have changed this incorrectly, actually. I think I need to change the request to `RepositoryHookRequest` instead.